metrics/web/test/common/presentation/navigation/state/navigation_notifier_test.dart

```diff
@@ -247,6 +273,92 @@ void main() {
       },
     );
 
+    test(
+      ".handlePageParametersUpdates() does nothing if the given page parameters model is null",
+      () {
+        when(pageParametersFactory.create(any)).thenReturn(pageParametersModel);
+
+        notifier.handleAuthenticationUpdates(isLoggedIn: true);
+        notifier.push(DefaultRoutes.dashboard);
+
+        final initialPageParameters = notifier.currentPageParameters;
+
+        notifier.handlePageParametersUpdates(null);
+
+        expect(notifier.currentPageParameters, equals(initialPageParameters));
+      },
+    );
+
+    test(
+      ".handlePageParametersUpdates() does nothing if the given page parameters model is equal to the current page parameters",
+      () {
+        when(pageParametersFactory.create(any)).thenReturn(pageParametersModel);
+
+        notifier.handleAuthenticationUpdates(isLoggedIn: true);
+        notifier.push(DefaultRoutes.dashboard);
+
+        verifyNever(routeConfigurationLocationConverter.convert(any));
+      },
+    );
+
+    test(
+      ".handlePageParametersUpdates() updates a current page parameters model with the given one",
+      () {
+        notifier.handlePageParametersUpdates(pageParametersModel);
+
+        expect(notifier.currentPageParameters, equals(pageParametersModel));
+      },
+    );
+
+    test(
+      ".handlePageParametersUpdates() updates a current configuration with the given page parameters",
+      () {
+        final expectedConfiguration = notifier.currentConfiguration.copyWith(
+          parameters: parameters,
```

Comment:
🔨 
```suggestion
          parameters: pageParametersModel.toMap(),
```
Would it be a bit more clear?